{"version":3,"sources":["../../app-src/util/ApplicationException.js"],"names":["ApplicationException","Error","constructor","msg","name","exception","isApplicationException","err","Object","getPrototypeOf","getExceptionMessage","message","console","log"],"mappings":";;;;;;UAAaA,oB,GAAN,MAAMA,oBAAN,SAAmCC,KAAnC,CAAyC;;AAE9CC,oBAAYC,MAAM,EAAlB,EAAsB;;AAEpB,gBAAMA,GAAN;AACA,eAAKC,IAAL,GAAY,KAAKF,WAAL,CAAiBE,IAA7B;AACD;AAN6C,O;;;;AAShD,YAAMC,YAAYL,oBAAlB;;AAEO,eAASM,sBAAT,CAAgCC,GAAhC,EAAqC;;AAE1C,eAAOA,eAAeF,SAAf,IAA4BG,OAAOC,cAAP,CAAsBF,GAAtB,aAAsCF,SAAzE;AACD;;;;AAEM,eAASK,mBAAT,CAA6BH,GAA7B,EAAkC;;AAEvC,YAAID,uBAAuBC,GAAvB,CAAJ,EAAiC;AAC/B,iBAAOA,IAAII,OAAX;AACD,SAFD,MAEO;AACLC,kBAAQC,GAAR,CAAYN,GAAZ;AACA,iBAAO,uCAAP;AACD;AACF","file":"ApplicationException.js","sourcesContent":["export class ApplicationException extends Error {\r\n\r\n  constructor(msg = '') {\r\n\r\n    super(msg);\r\n    this.name = this.constructor.name;\r\n  }\r\n}\r\n\r\nconst exception = ApplicationException;\r\n\r\nexport function isApplicationException(err) {\r\n\r\n  return err instanceof exception || Object.getPrototypeOf(err) instanceof exception\r\n}\r\n\r\nexport function getExceptionMessage(err) {\r\n\r\n  if (isApplicationException(err)) {\r\n    return err.message;\r\n  } else {\r\n    console.log(err);\r\n    return 'Não foi possível realizar a operação.';\r\n  }\r\n}"]}